### ISSUE TITLE
5-Compositional Visual Generation with Composable Diffusion Models

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#5]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by geekn0rd** [comment]
### **Introduction and Objective:**
Large text-guided diffusion models, such as Stable Diffusion, are able to generate  photorealistic images given natural language prompts. While such models are highly flexible, they struggle to understand the composition of certain concepts, such as confusing the attributes of different objects or relations between objects.
An underlying difficulty may be that such models encode text descriptions as fixed-size latent vectors and when text descriptions become more and more complex they can not be squeezed into a fixed size vector. 
This paper suggests that instead of a long, complicated description, it is better to divide it into shorter descriptions that can be joined together, using AND and NOT operators.  During inference, they utilize multiple diffusion models that each take care of one specific concept and then combine their results.


###  **Methodology:**
The sampling procedure of diffusion models in and EBMs in are functionally similar.
 We may view a trained diffusion model $\epsilon_θ(x_t, t)$ as an implicitly parameterized EBM. Such parameterization enables us to apply previous techniques for composing EBMs to diffusion models. We aim to generate images conditioned on a set of concepts $\{c_1, c_2, . . . , c_n\}$.
**Compositional generation:** This method can compose multiple diffusion models during inference and generate images containing all the concepts described in the inputs without further training. It first sends an image from iteration $t$ and each individual concept $c_i$ to the diffusion model to generate a set of scores $\{ \epsilon_{\theta}(x_t, t | c_1), ..., \epsilon_{\theta}(x_t, t | c_n) \}$. They then compose different concepts using the proposed compositional operators, such as conjunction, to denoise the generated images. The final image is obtained after $T$ iterations.
**Concept Conjunction (AND):** 
We sample from this resultant distribution with the composed score function:
$\hat\epsilon(x_t, t) = \epsilon_{\theta}(x_t, t)  + \sum w_i((\epsilon_{\theta}(x_t, t|c_i)) - \epsilon_{\theta}(x_t, t)$)
 where $w_i$ is a hyperparameter corresponding to the temperature scaling on concept $c_i$. In the setting in which image generation is conditioned on a single concept, the above sampling procedure reduces to the classifier-free guidance
 **Concept Negation (NOT):** 
In concept negation, we aim to generate realistic images with the absence of a certain factor $\tilde{c_j}$. However, the negation of a concept can be ill-defined. Thus we generate images conditioned on other concepts as well to make the generated images look real.
$\hat\epsilon(x_t, t) = \epsilon_{\theta}(x_t, t)  + w((\epsilon_{\theta}(x_t, t|c_i)) - \epsilon_{\theta}(x_t, t|\tilde{c_j})$)
where $w$ is the hyperparameter that controls the strength of the negation.

We can generate images with the composed concepts using the following generative process:
$p_{compose}(x_{t−1}|x_t):=\mathcal{N} (x_t − \hat\epsilon(x_t, t) 
 ,{σ_t}^2 I)$
###  **Contributions:**
- They show that diffusion models can be composed by interpreting them as energy-based models, and drawing on this connection, they demonstrate how to compose diffusion models together and generate images from more complex text prompts.

- They propose two compositional operators, Conjunction and Negation, on top of diffusion models that allow they to compose concepts in different domains during inference without any additional training.

- Finally, we evaluate our method on composing language descriptions, objects, object relations, and human facial attributes.

###  **Experiments/Results:**
- They use CLEVR, Relational CLEVR and FFHQ datasets.
- For each task, They use the training data (real images) to train a binary classifier that takes an image and a concept, e.g. ‘smiling’, as input, and predicts whether the image contains or represents the concept. They then apply this classifier to a generated image, checking whether it faithfully captures each of the concepts.
- Frechet Inception Distance (FID) is a commonly used metric for evaluating the quality of generated images. It uses a pre-trained inception model to extract features for the generated images and real images, and measures their feature similarity.

### **Comparison:**
 - They develop Composed GLIDE, a version of GLIDE that utilizes the compositional operators to combine textual descriptions, without further training. They compare it to the original GLIDE, which directly encodes the descriptions as a single long sentence. Their approach more accurately captures text details, such as the "overwater bungalows" in the third example.
![glide](https://energy-based-model.github.io/Compositional-Visual-Generation-with-Composable-Diffusion-Models/website_files/imgs/glide.png)
###  **Conclusion and Future Work:**
- By interpreting diffusion models as energy-based models, we may explicitly compose them and generate images with significantly more complex combinations that are never seen during training.
 - The proposed composable diffusion models can generate images conditioned on sentence descriptions, objects, object relations, and human facial attributes, and can generalize to new combinations that are rarely seen in the real world.
 -  While the proposed composition operators is used for image generation, it may further applied to the generation of waveforms, 3D shapes, decision making, and text.
 - We found limited success when composing diffusion models trained on different datasets. In contrast, compositional generation with EBMs can successfully compose multiple separately trained models. Incorporating additional structures into diffusion models from EBMs such as a conservative score field, can be a promising direction towards compositions of separately trained diffusion models.
